jupyterhub_ssh/__init__.py

```diff
@@ -221,17 +222,37 @@ class JupyterHubSSH(Application):
         config=True,
     )
 
-    # FIXME: Make this a yarl.URL object?
     hub_url = Any(
         "",
         help="""
-        URL of JupyterHub to connect to.
+        URL of JupyterHub's proxy to connect to.
+        
+        `jupyterhub-ssh` needs to be able to connect to both the
+        JupyterHub API and to the user's server. If running in
+        the same cluster as the JupyterHub, this should point to
+        the `proxy-public` service. When running outside the same
+        cluster, this should be set to the user accessible URL of
+        the hub.
 
         *Must* be set.
+        
+        Examples:
+        
+        `http://proxy-public` (in same cluster & namespace as jupyterhub)
+        `https://my-hub-url.com` (when running outside the cluster)
```

Comment:
```suggestion
        URL of JupyterHub's proxy to connect to.

        jupyterhub-ssh needs to be able to connect to both the JupyterHub API
        (/hub/api) and to the users' servers (/user/<username>) via HTTP or
        HTTPS. This URL doesn't have to be the public URL though as the URL is
        only be used by jupyterhub-ssh itself.

        *Must* be set.

        Examples:

        - If jupyterhub-ssh is deployed in the same Kubernetes cluster and
          namespace as the official JupyterHub Helm chart, you can use
          `http://proxy-public:8000` or `http://proxy-http:8000` depending on
          how the JupyterHub Helm chart is configured. Use
          `http://proxy-http:8000` if the JupyterHub Helm chart has been
          configured with `proxy.https.enabled=true` and
          `proxy.https.type=letsencrypt`.

        - If jupyterhub-ssh can't access JupyterHub's proxy via local network or
          you don't trust the local network to be secure, use a public URL with
          HTTPS such as `https://my-hub-url.com`.
```

Reply:
Hmm, in what cases should it be proxy-http? And proxy-public exposes port 80, not 8000, right?

Reply:
`proxy-http` should be used when autohttps pod is active and proxy-public redirects to it, which in turn redirects to https as traefik is configured.

Reply:
The autohttps pod in turn is used when 


the JupyterHub Helm chart has been
          configured with `proxy.https.enabled=true` and
          `proxy.https.type=letsencrypt`.

Reply:
Is that just to avoid an extra hop? Do you think you will be able to test it? If it works, let's explicitly specify that?

Reply:
proxy-http:8000 works for me in JMTE, if we try to use use proxy-public when autohttps is enabled, I'm very confident that we have a "redirect middleware" that makes all HTTP traffic be redirected to the HTTPS path.

If such redirection happen, it is configured to redirect to HTTPS:// protocol of the same domain etc, but that makes the request fail because https://proxy-public would error based on invalid certificate. Due to this, proxy-public can't be used, but instead we need to use proxy-http:8000 that will accept HTTP without forcing HTTPS.

Reply:
> Do you think you will be able to test it? If it works, let's explicitly specify that?

I'm a bit unsure what you refer to with `it`. I perceive it as the suggested edit does specify some things of relevance explicitly.

```
        - If jupyterhub-ssh is deployed in the same Kubernetes cluster and
          namespace as the official JupyterHub Helm chart, you can use
          `http://proxy-public` or `http://proxy-http:8000` depending on
          how the JupyterHub Helm chart is configured. Use
          `http://proxy-http:8000` if the JupyterHub Helm chart has been
          configured with `proxy.https.enabled=true` and
          `proxy.https.type=letsencrypt`.
```

I'll update the code suggestion with an "otherwise" statement saying "otherwise use proxy-public"